Bypasses hardcoded host-system hardware checks for memory.

Original issue: RCCL uses raw bash commands to check host RAM and dynamically calculate
the number of parallel linker jobs. In EasyBuild environment, these commands often fail
or miscalculate, leading to CMake errors or OOM crashes during the linking phase.

This patch explicitly limits `num_linker_jobs` to `4` to ensure stable compilation.

For ROCm version, use EXPLICIT_ROCM_VERSION CMake variable (already supported in standalone RCCL).

Author: Aayush Joglekar (aayush.joglekar@surf.nl)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 45ba954f..0d631a2a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1304,26 +1304,9 @@ endif()

 ## Set RCCL link options
 ## Find out available memory
-execute_process(
-  COMMAND bash "-c" "cat /sys/fs/cgroup/memory.max"
-  OUTPUT_VARIABLE memory_max_string)
-if (${memory_max_string} MATCHES "^[0-9]+")
-  math(EXPR memory_in_gb "${memory_max_string} / (1024 * 1024 * 1024)")
-else()
-  execute_process(
-    COMMAND bash "-c" "free | grep -o '[[:digit:]]*' | head -1"
-    OUTPUT_VARIABLE memory_max_string)
-  ## memory_max_string holds the free memory in KB
-  if (${memory_max_string} MATCHES "^[0-9]+")
-    math(EXPR memory_in_gb "${memory_max_string} / (1024 * 1024)") ## KB to GB conversion
-  else()
-    cmake_host_system_information(RESULT memory_max_string QUERY AVAILABLE_PHYSICAL_MEMORY )
-    math(EXPR memory_in_gb "${memory_max_string} / 1024")
-  endif()
-endif()
+set(num_linker_jobs "4")
 ## Reserve 16GB for each linker job. Limit max number of linker jobs to 16
 if (HAVE_PARALLEL_JOBS)
-  math(EXPR num_linker_jobs "(${memory_in_gb} + 15) / 16")
   if (${num_linker_jobs} GREATER_EQUAL "16")
     set(num_linker_jobs "16")
   endif()
